ctapipe/image/muon/muon_ring_finder.py

```diff
@@ -1,28 +1,139 @@
 import numpy as np
 import astropy.units as u
-from ctapipe.image.muon.ring_fitter import RingFitter
+from ctapipe.core import Component
 from ctapipe.io.containers import MuonRingParameter
+import traitlets as traits
+from iminuit import Minuit
+
+__all__ = ['MuonRingFitter']
+
+
+class MuonRingFitter(Component):
+    """Different ring fit algorithms for muon rings
+    """
+    fit_method = traits.CaselessStrEnum(
+        ['taubin', 'chaudhuri_kundu'],
+        default_value='chaudhuri_kundu'
+    ).tag(config=True)
+
+    tel_type = traits.CaselessStrEnum(
+        ['LST_LST_LSTCam', 'MST_MST_NectarCam', 'MST_MST_FlashCam', 'MST_SCT_SCTCam',
+         'SST_1M_DigiCam', 'SST_GCT_CHEC', 'SST_ASTRI_ASTRICam', 'SST_ASTRI_CHEC'],
+        default_value='LST_LST_LSTCam'
+    ).tag(config=True)
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config, parent, **kwargs)
+
+    def fitFormula(self, xc, yc, r):
+        '''taubin fit formula
+        reference : Barcelona_Muons_TPA_final.pdf (slide 6)
+        '''
+        upper_term = sum(
```

Comment:
using python's sum is slow on numpy array, fastest is to use the `.sum()` method.

So do: `upper_term = ((x - xc)**2 + (y - yc)**2 - r**2).sum()`